### PR TITLE
[APK] impl refactor for APK

### PR DESCRIPTION
**Pull request by chameerar**
Refactoring changes in REST API impl for APK

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/publisher/SearchApiServiceImplUtil.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (RakhithaRR) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/publisher/SearchApiServiceImplUtil.java:2]: This has to WSO2 LLC. Check other places as well.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/publisher/ApisApiServiceImplUtils.java [hunk 2]
```diff
@@ -70,9 +74,7 @@
 import java.net.URL;
 import java.net.UnknownHostException;
 import java.nio.charset.StandardCharsets;
-import java.util.Collections;
-import java.util.List;
-import java.util.Set;
+import java.util.*;
```
Comment (RakhithaRR) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/publisher/ApisApiServiceImplUtils.java:77]: Let's reset those wildcard imports

**Review by RakhithaRR — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/publisher/SearchApiServiceImplUtil.java (hunk 1 above)
Comment (chameerar) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/publisher/SearchApiServiceImplUtil.java]: Resolved in a05bcd7fc00248a0d25d1496cf127e5caee4cdfc

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/publisher/ApisApiServiceImplUtils.java (hunk 2 above)
Comment (chameerar) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/publisher/ApisApiServiceImplUtils.java]: Resolved in a05bcd7fc00248a0d25d1496cf127e5caee4cdfc

**Review by tharindu1st — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/publisher/ApisApiServiceImplUtils.java [hunk 3]
```diff
@@ -669,4 +685,266 @@ public static API importAPIDefinition(API apiToAdd, APIProvider apiProvider, Str
 
         return addedAPI;
     }
+
+    /**
+     * @param api API
+     * @param soapOperation SOAP Operation
+     * @return SOAP API Definition
+     * @throws APIManagementException if an error occurred while parsing string to JSON Object
```
Comment (wasuradananjith) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/publisher/ApisApiServiceImplUtils.java:693]: Javadoc comments are not properly formatted. Shall we fix those?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/publisher/ApisApiServiceImplUtils.java [hunk 4]
```diff
@@ -669,4 +685,266 @@ public static API importAPIDefinition(API apiToAdd, APIProvider apiProvider, Str
...
+    public static String getSwaggerString(InputStream fileInputStream, String url, String wsdlArchiveExtractedPath,
+                                          String filename) throws APIManagementException, IOException {
+
+        String swaggerStr = "";
+        if (StringUtils.isNotBlank(url)) {
+            swaggerStr = SOAPOperationBindingUtils.getSoapOperationMappingForUrl(url);
+        } else if (fileInputStream != null) {
+            if (filename.endsWith(".zip")) {
+                swaggerStr = SOAPOperationBindingUtils.getSoapOperationMapping(wsdlArchiveExtractedPath);
+            } else if (filename.endsWith(".wsdl")) {
```
Comment (wasuradananjith) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/publisher/ApisApiServiceImplUtils.java:731]: Can't we use a constant [1] here?

[1] https://github.com/wso2/carbon-apimgt/blob/834c40e6400554a39b7835a45b7ccea681387e2c/apk/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/APIConstants.java#L185

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/publisher/ApisApiServiceImplUtils.java [hunk 5]
```diff
@@ -669,4 +685,266 @@ public static API importAPIDefinition(API apiToAdd, APIProvider apiProvider, Str
...
+     * @throws IOException If error occurred in converting InputStream to a byte array
+     */
+    public static String getSwaggerString(InputStream fileInputStream, String url, String wsdlArchiveExtractedPath,
+                                          String filename) throws APIManagementException, IOException {
+
+        String swaggerStr = "";
+        if (StringUtils.isNotBlank(url)) {
+            swaggerStr = SOAPOperationBindingUtils.getSoapOperationMappingForUrl(url);
+        } else if (fileInputStream != null) {
+            if (filename.endsWith(".zip")) {
```
Comment (wasuradananjith) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/publisher/ApisApiServiceImplUtils.java:729]: Shall we use a constant [1] here?

[1] https://github.com/wso2/carbon-apimgt/blob/834c40e6400554a39b7835a45b7ccea681387e2c/apk/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/APIConstants.java#L1736

**Pull request merged**